frontend/packages/kubevirt-plugin/src/components/vm-nics/vm-nics.tsx

```diff
@@ -108,9 +115,36 @@ export const VMNicsTable: React.FC<VMNicsTableProps> = ({
   );
 };
 
-export const VMNics: React.FC<VMLikeEntityTabProps> = ({ obj: vmLikeEntity }) => {
+type VMNicsProps = {
+  obj: VMGenericLikeEntityKind;
+  vmis?: FirehoseResult<VMIKind[]>;
+  updatePendingChanges?: (pc: PendingChanges) => void;
+};
+
+export const VMNics: React.FC<VMNicsProps> = ({
+  obj: vmLikeEntity,
+  vmis,
+  updatePendingChanges,
+}) => {
   const [isLocked, setIsLocked] = useSafetyFirst(false);
   const withProgress = wrapWithProgress(setIsLocked);
+
+  const loadedVMIs = vmis && getLoadedData(vmis);
+  const vmi = loadedVMIs && loadedVMIs.length > 0 && loadedVMIs[0];
+  const isVMRunning = isVM(vmLikeEntity) && isVMRunningOrExpectedRunning(vmLikeEntity);
+
+  if (isVMRunning) {
+    updatePendingChanges({
```

Comment:
why does these have to be called from every component?
Can't the pending-changes component handle all these by itself?

plus there are couple of problems with this
- not a react way (it is very bad to run functions/side effects inside a render)
- is there a reason a control should flow from bottom up?
- if so such behaviour should be declarative and not functional